### PR TITLE
[RSDK-7223] move-camera-rtp-passthrough-to-own-package

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/RSDK-7223)

Fixes build issue introduced in: https://github.com/viamrobotics/rdk/pull/3723

Moves RTP Passthrough features into submodule of camera so that building the viam CLI doesn't fail due to the camera package pulling in unneeded C deps as a side effect of module depending on the rtp passthrough interface.

I also renamed some types to shorten names & make it clearer that these features are about RTP passthrough. 

Cameras supporting RTP passthrough will need to implement the `camera/rtppassthrough.Source` interface and no longer need to implement the `VideoCodecStreamSource` method. 

Tested for compatibility with old modules & passthrough modules with:
```json
{
  "modules": [
    {
      "name": "rtsp-nick",
      "executable_path": "/Users/nicksanford/code/viamrtsp/bin/viamrtsp",
      "type": "local"
    },
    {
      "executable_path": "/Users/nicksanford/code/robotModules/new/modmain",
      "name": "nicks-fake-new",
      "type": "local"
    },
    {
      "executable_path": "/Users/nicksanford/code/robotModules/old/modmain",
      "name": "nicks-fake-old",
      "type": "local"
    },
    {
      "module_id": "viam-labs:overlay-fps",
      "name": "viam-labs_overlay-fps1",
      "type": "registry",
      "version": "0.0.4"
    }
  ],
  "components": [
    {
      "attributes": {
        "rtp_passthrough": true,
        "rtsp_address": "rtsp://localhost:8554/live1"
      },
      "depends_on": [],
      "name": "rtsp-passthrough",
      "model": "erh:viamrtsp:rtsp-h264",
      "type": "camera",
      "namespace": "rdk"
    },
    {
      "depends_on": [
        "rtsp-passthrough"
      ],
      "model": "viam-labs:camera:overlay-fps",
      "name": "fps-h2641",
      "namespace": "rdk",
      "type": "camera",
      "attributes": {
        "camera_name": "rtsp-passthrough"
      }
    },
    {
      "attributes": {
        "rtsp_address": "rtsp://localhost:8554/live1"
      },
      "depends_on": [],
      "name": "rtsp",
      "model": "erh:viamrtsp:rtsp-h264",
      "type": "camera",
      "namespace": "rdk"
    },
    {
      "depends_on": [],
      "model": "nicks-old:camera:fake",
      "name": "nicks-old",
      "namespace": "rdk",
      "type": "camera",
      "attributes": {
        "animated": true
      }
    },
    {
      "depends_on": [],
      "model": "nicks-new:camera:fake",
      "name": "nicks-new",
      "namespace": "rdk",
      "type": "camera",
      "attributes": {
        "animated": true
      }
    },
    {
      "depends_on": [],
      "model": "rdk:builtin:fake",
      "name": "local",
      "namespace": "rdk",
      "type": "camera",
      "attributes": {
        "animated": true
      }
    }
  ]
}
```

### CODE REVIEW

File: components/camera/camera.go [hunk 1]
```diff
@@ -82,22 +82,12 @@ type Camera interface {
...
-		SubscribeRTP(ctx context.Context, bufferSize int, packetsCB PacketCallback) (StreamSubscriptionID, error)
-		Unsubscribe(ctx context.Context, id StreamSubscriptionID) error
-	}
-)
-
 // A VideoSource represents anything that can capture frames.
 type VideoSource interface {
 	projectorProvider
-	// A VideoCodecStreamSource allows modules to passthrough RTP packets over used by viam-server's camera client to stream video over webrtc.
+	// A RTPPassthroughSource allows modules to passthrough RTP packets over used by viam-server's camera client to stream video over webrtc.
```
Comment (randhid) [inline components/camera/camera.go:88]: love love love love the name change

**Review by randhid — APPROVED** [review]
Thank you for the name changes as well, they're more descriptive.

**Review by dgottlieb — APPROVED** [review]
My skim of the diff does confirm that the change is purely a refactor as the description implied.

File: components/camera/camera.go [hunk 2]
```diff
@@ -138,20 +126,44 @@ type ImagesSource interface {
...
 
 type sourceBasedCamera struct {
 	resource.Named
 	resource.AlwaysRebuild
 	VideoSource
+	rtpPassthroughSource rtppassthrough.Source
 	logging.Logger
 }
 
+func (vs *sourceBasedCamera) SubscribeRTP(
```
Comment (nicksanford) [inline components/camera/camera.go:149]: This is needed for cameras that  are wrapped in `FromVideoSource` to still implement `rtppassthrough.SubscriptionID`

File: components/camera/camera.go [hunk 3]
```diff
@@ -189,20 +201,31 @@ func NewVideoSourceFromReader(
...
+		videoStream:          gostream.NewEmbeddedVideoStream(vs),
+		actualSource:         reader,
+		imageType:            imageType,
 	}, nil
 }
 
-func (vs *videoSource) VideoCodecStreamSource(ctx context.Context) (VideoCodecStreamSource, error) {
-	if vs.videoCodecStreamSource != nil {
-		return vs.videoCodecStreamSource, nil
+func (vs *videoSource) SubscribeRTP(
```
Comment (nicksanford) [inline components/camera/camera.go:213]: This is needed for cameras that are wrapped in `videoSource` to still implement `rtppassthrough.SubscriptionID`

**Pull request merged**